src/controllers/post.controller.ts

```diff
@@ -10,12 +10,24 @@ class PostController {
     try {
       const userId = req.user.id
       const postData: CreatePostRequestDto = req.body;
-      const post : BasePostDto = await this.postService.createPosts({...postData, userId});
-      res.status(201).json({ post, message: 'success' });
+      const data : BasePostDto = await this.postService.createPosts({...postData, userId});
+      res.status(200).json({ data, message: 'Success' });
     } catch (error) {
       next(error);
     }
   };
-}
 
-export default PostController;
+  public updatePost = async (req: RequestWithUser, res: Response, next: NextFunction) => {
+    try {
+      const userId = req.user.id;
+      const updatedPostData: UpdatePostRequestDto = req.body;
+      const id = Number(req.params.post_id);
+      const update: BasePostDto = await this.postService.updatePost(id, { ...updatedPostData }, userId);
```

Comment:
nit: we can just do this `this.postService.updatePost(id, updatedPostData, userId)`

Reply:
Also let's update the variable to be called `data` so we can 
`res.status(200).json({ data, message: 'Success' });` so it's easier to work with in the frontend